_posts/en/newsletters/2023-02-08-newsletter.md

```diff
@@ -54,12 +54,80 @@ Bitcoin infrastructure software.
 meeting, highlighting some of the important questions and answers.  Click on a
 question below to see a summary of the answer from the meeting.*
 
-FIXME is a PR by FIXME that FIXME:LarryRuane
+[Track AddrMan totals by network and table, improve precision of adding fixed seeds][review club 26847]
+is a PR by Martin Zumsande, co-authored by Amiti Uttarwar, that
+allows the Bitcoin Core client to more reliably find outbound peers
+and distribute them more broadly across available network types (for
+example, IPV4, IPV6, tor, i2p). It does this by enhancing `AddrMan`
+(the peer address manager) to keep track of the number of address entries
+separately by network and "tried" versus "new" type, which in turn allows
+better use of fixed seeds. This is the first step in a larger effort to
+improve outbound peer selection.
 
 {% include functions/details-list.md
-  q0="FIXME"
-  a0="FIXME"
-  a0link="https://bitcoincore.reviews/FIXME#l-FIXME"
+  q0="When is a network considered reachable?"
+  a0="A network is assumed to be reachable unless we're sure we
+      can't access it, or our configuration has specified one or
+      more _other_ networks using the `-onlynet=` configuration option
+      (only those are considered reachable, even if other network
+      types are actually available).
+  a0link="https://bitcoincore.reviews/26847#l-22"
+
+  q1="How is an address received over the p2p network treated depending
+      on whether the address's network is reachable vs. non-reachable --
+      do we store it (add it to `AddrMan`) and/or forward it to peers?"
+  a1="If its network is reachable, we relay the address to two
+      randomly-chosen peers, else we relay it to 1 or 2 peers (whether
+      1 or 2 is randomly-chosen).
+      We only store the address if its network is reachable."
+  a1link="https://bitcoincore.reviews/26847#l-51"
+
+  q2="How can a node currently get stuck with only unreachable addresses
+      in `AddrMan`, finding no outbound peers? How does this PR fix it?"
+  a2="If the `-onlynet` configuration changes. For example, suppose the
+      node has always been run with `-onlynet=ipv4` so its AddMan has
+      no ipv6 addresses. Then the node is restarted with `-onlynet=ipv6`.
```

Comment:
I would replace ipv6 with onion or i2p in this example. Switching between clearnet and a privacy network makes much more sense as a use case.

Reply:
> I would replace ipv6 with onion or i2p in this example. Switching between clearnet and a privacy network makes much more sense as a use case.

FWIW I believe neither of these would be typical real-world cases and that most onlynet users are either setting onlynet=onion, or since a couple of years now, also onlynet=onion and i2p.

Reply:
Oh, just saw your comment here moments after my latest push... Is it worth fixing, do you think?

Reply:
I think it's more credible to use a real world case but it's just a suggestion.

Reply:
Not sure if this discussion belongs here - but I think it's quite possible that a user would do IBD with a node in default clearnet mode (without setting up tor, so that Tor is unreachable), and after finishing IBD, they would install/activate tor and decide to switch to `-onlynet=tor`. This is now possible without requiring an intermediate period in which they'd need to be connected to multiple networks or requiring any manual steps, like deleting `peers.dat` or setting up manual connections.

Reply:
Agree. Though once the node has peers in its reachable networks, manual connections are likely the best way to ensure it remains connected to at least one honest peer, more so with Tor and I2P if it's easier for an attacker to spin up many nodes.  I should probably update doc/{tor,i2p,cjdns}.md (and my blog) to communicate this.

Reply:
> I think it's quite possible that a user would do IBD with a node in default clearnet mode (without setting up tor, so that Tor is unreachable), and after finishing IBD, they would install/activate tor and decide to switch to `-onlynet=onion`.

Just as a point of information, that's _exactly_ with I did with the little RPi [myNode](https://mynodebtc.com/) when I bought it last year. I started the IBD over default onion, and it was going so slowly that it was going to take months (rough estimate). I switched to clearnet and it finished in about a week (still slow, but it's a tiny CPU). Then I switched back to the default onion. I probably had some peers in addrman left over from the first phase? I didn't notice it getting stuck when I switched back to onion.